docs/docs/tutorials/providers.md

```diff
@@ -0,0 +1,561 @@
+
+So, if everything is set up properly, you should see the transaction receipt and a the new head printed in the console.
+
+IPC provider is a websocket alternative when web3 needs to communicate with the Ethereum nodes over `Inter-process Communication`. The IPC providers, similar to WebSocket providers, support subscriptions, which means we can subscribe to contract events and receive real-time updates.
+
+Keep in mind that using IPC Provider with `geth` in development mode in a production environment is not recommended as it can pose a security risk.
+
+**IPC Provider (for Node.js)**
+
+IPC Provider allows us to interact with the Ethereum network via Inter-Process Communication (IPC), which is useful for Node.js applications that require direct access to local Ethereum node. This provider is ideal for development and testing on a local machine.
+
+To connect to Ethereum network using IPC Provider through `geth` in development mode, you will need to run `geth` with the `--dev --ipcpath <path>` command line arguments:
+
+1. Start a `geth` node in development mode by opening a terminal window and navigating to the `geth` executable file. Then, run the following command to create a development chain:
+
+```sh
+geth --dev --ipcpath <path>
+```
+
+Make sure to replace `<path>` with the desired IPC path. For example:
+
+```sh
+geth --dev --ipcpath /Users/username/Library/Ethereum/geth.ipc
+```
+
+This will start a `geth` node in development mode with IPC enabled and an IPC path specified. If the command is successful, the `geth` node will be running, and you should see output similar to the following:
+
+```sh
+INFO [12-10|15:10:37.121] IPC endpoint opened              url=<path>
+INFO [12-10|15:10:37.122] HTTP endpoint opened             url=http://localhost:8545
+INFO [12-10|15:10:37.122] WebSocket endpoint opened        url=ws://localhost:8546
+INFO [12-10|15:10:37.127] Mapped network port              proto=udp extport=0 intport=30303 interface=UPnP(UDP)
+```
+
+2. Open a command prompt or terminal window and navigate to where you would like to create the folder for this example.
+3. Create a new folder and navigate to it:
+
+    ```sh
+    mkdir web3-providers-tutorial
+    cd web3-providers-tutorial
+    ```
+
+4. Install Web3.js using npm:
+
+    ```sh
+    npm install web3
+    ```
+
+5. Create a new JavaScript file called `web3-ipc-provider.js` in your code editor.
+6. Copy and paste the following code into your `web3-ipc-provider.js` file and save it:
+
+    ```js
+    const { Web3 } = require('web3');
+
+    // Connect to the Ethereum network using IPC provider
+    const ipcPath = '<path>';
+    const ipcProvider = new Web3.providers.IpcProvider(ipcPath);
```

Comment:
this code will need update as IPC is not exported through main web3 package now.